src/process.rs

```diff
@@ -142,3 +147,96 @@ impl<T: Runtime> Future for ProcessController<T> {
         }
     }
 }
+
+pub struct LossyLinesCodec {
+    max_length: usize,
+}
+
+impl Default for LossyLinesCodec {
+    fn default() -> Self {
+        Self {
+            max_length: usize::MAX,
+        }
+    }
+}
+
+pub type OutputLines = Pin<Box<dyn futures::Stream<Item = anyhow::Result<String>> + Send>>;
+
+/// Reads process stdout and stderr using `LossyLinesCodec`
+pub fn process_output(child: &mut Child) -> anyhow::Result<OutputLines> {
+    let stdout = child
+        .stdout
+        .take()
+        .context("Failed to access process stdout")?;
+    let stderr = child
+        .stderr
+        .take()
+        .context("Failed to access process stderr")?;
+
+    let stdout = FramedRead::new(BufReader::new(stdout), LossyLinesCodec::default());
+    let stderr = FramedRead::new(BufReader::new(stderr), LossyLinesCodec::default());
+
+    Ok(futures::StreamExt::boxed(stdout.merge(stderr)))
+}
+
+/// Decodes lines as UTF-8 (lossly) up to `max_length` characters per line.
+impl Decoder for LossyLinesCodec {
+    type Item = String;
+
+    type Error = anyhow::Error;
+
+    fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        let read_to = std::cmp::min(self.max_length.saturating_add(1), buf.len());
+        let new_line_offset = buf[0..read_to].iter().position(|b| *b == b'\n');
+        let has_new_line = new_line_offset.is_some();
+        let offset = new_line_offset.unwrap_or(read_to);
+        let line = buf.split_to(offset);
+        if has_new_line {
+            // Move cursor pass new line character so next call of `decode` will not read it.
+            buf.advance(1);
+        }
+        let mut line: &[u8] = &line;
+        if let Some(&b'\r') = line.last() {
+            // Skip carriage return.
+            line = &line[..line.len() - 1];
+        }
+        if line.is_empty() {
+            return Ok(None);
+        }
+        // Lossy conversion to avoid errors when converting UTF-16 characters on Windows process output.
```

Comment:
```suggestion
        // Process output on Windows is encoded in UTF-16. To avoid OS specific implementation of process output handling the output is lossly converted to UTF-8. 
        // It allows to avoid errors when decoding some Windows error log messages.
```